proto/entities/v1alpha/entities.proto

```diff
@@ -22,6 +24,16 @@ service Entities {
   // For groups and documents, we match the title, while we match alias in accounts.
   rpc SearchEntities(SearchEntitiesRequest) returns (SearchEntitiesResponse);
 
+  // Deletes an entity from the local node. It removes all the patches corresponding to it, including comments.
+  rpc DeleteEntity(DeleteEntityRequest) returns (google.protobuf.Empty);
+
+  // Lists deleted entities.
+  rpc ListDeletedEntities(ListDeletedEntitiesRequest) returns (ListDeletedEntitiesResponse);
+
+  // Try to bring back the deleted entity in next syncync round. It may fail if there is no
+  // provider for that entity, even though it once was when the entity was created.
+  rpc RestoreEntity(RestoreEntityRequest) returns (google.protobuf.Empty);
```

Comment:
Does restore perform the discovery all in one go? I think restore should just remove the entry from the deleted table, but the client should call the content discovery. This would let you handle the feedback from the discovery process better.

Also, if this call would fail because discovery didn't work, the item would still be removed from the deleted table, right? That would be somewhat confusing, because it feels like the process ended up in a "partial" state.

Reply:
Actually this is poor documented. It does not do what it says it does. What really does is remove an entry from the `deleted_resources` table. Nothing more. So the frontend should call this function and then discoverentity. The name `RestoreEntty` is confusing and the comments as well. How should I name them? `RemoveDeletedEntity`? because removing from a deletion table is also confusing I did not know how to name it Any suggestions @burdiyan ?

Reply:
@juligasa Maybe "UndeleteEntity"? https://google.aip.dev/164#undelete

We should clearly document that it doesn't actually restore, but let's the item appear again if synced or discovered.